quinn-proto/src/endpoint.rs

```diff
@@ -575,12 +575,17 @@ impl Endpoint {
             return Err(TransportError::CONNECTION_REFUSED(""));
         }
 
-        if dst_cid.len() < 8
-            && (!server_config.use_retry || dst_cid.len() != self.local_cid_generator.cid_len())
+        // RFC9000 §7.2 dictates that initial (client-chosen) destination CIDs must be at least 8
+        // bytes. If this might be a Retry packet, then the length may instead match our usual CID
```

Comment:
"If this might be" sounds a bit strange to me? Maybe just if "If this is" or "In case this is"?